resources/wikia/modules/tracker.js

```diff
@@ -135,15 +135,19 @@
 		 */
 		function sendInternalTrackingEvent( event, data, timeout ) {
 			var head = document.head || document.getElementsByTagName( 'head' )[ 0 ] || document.documentElement,
-					script = document.createElement( 'script' ),
-					requestUrl = 'https://beacon.wikia-services.com/__track/special/' + encodeURIComponent( event ),
-					requestParameters = [],
-					p,
-					params,
-					userId = -1;
+				script = document.createElement( 'script' ),
+				requestUrl = 'https://beacon.wikia-services.com/__track/special/' + encodeURIComponent( event ),
+				requestParameters = [],
+				p,
+				params,
+				userId = -1,
+				wikiVariables = M && M.getFromHeadDataStore && M.getFromHeadDataStore('wikiVariables'),
+				trackingDimensions = M && M.getFromHeadDataStore && M.getFromHeadDataStore('trackingDimensions');
 
 			if (isOptedIn) {
-				userId = window.wgTrackID;
+				userId = window.wgTrackID
+					|| (M && M.getFromHeadDataStore && parseInt(M.getFromHeadDataStore('userId'), 10))
```

Comment:
What do you think about moving it to L143

Reply:
I don't think it's much more readable, but yeah, why not. Moved.